src/tribler-common/tribler_common/dependencies.py

```diff
@@ -0,0 +1,107 @@
+"""
+This file lists the python dependencies for Tribler.
+
+Note that this file should not depend on any external modules itself other than builtin ones.
+"""
+import importlib
+import logging
+import re
+import sys
+from enum import Enum
+from typing import Iterator
+
+import pkg_resources
+
+from tribler_common.utilities import is_frozen, show_system_popup
+
+from tribler_core.utilities.install_dir import get_base_path
+from tribler_core.utilities.path_util import Path
+
+# fmt: off
+
+logger = logging.getLogger(__name__)
+
+Scope = Enum('Scope', 'core gui common')
+
+_frozen_env = is_frozen()
+_base_path = get_base_path().parent.parent  # this is the "/src" folder
```

Comment:
**issue:** `.parent.parent` affects the frozen path for requirements file.

Reply:
Thank you, fixed.

Also, I've added test for this case:

```python
async def test_dev_get_requirements_path():
    assert _get_requirements_path(scope=Scope.common, frozen=False) == Path(
        tribler_common.__file__).parent / 'requirements.txt'


@patch_dependencies_module(get_base_path, new=MagicMock(return_value=Path('')))
async def test_frozen_get_requirements_path():
    assert _get_requirements_path(scope=Scope.common, frozen=True) == Path('requirements-common.txt')
```